test/integration/canonical_test.rb

```diff
@@ -34,18 +34,33 @@ class CanonicalTest < ActionDispatch::IntegrationTest
       @post.site.update!(view_all: true)
     end
 
-    sub_test_case "category page" do
+    sub_test_case "category (with a post) page" do
       setup do
-        @category = create(:category, site: @post.site)
+        @category = @post.categories.first
```

Comment:
Can you use `first!` instead of `first`.
The former raises an error when category is nil.
It is better way to fail fast for unexpected data.